From f741744ecfc6bd2c3bc41cc727cbf822a85420b9 Mon Sep 17 00:00:00 2001
From: Martijn van de Streek <martijn@vandestreek.net>
Date: Sat, 4 Nov 2017 18:53:52 +0100
Subject: [PATCH 1/2] Add missing libraries for pcscd to hook script

This adds libusb-1.0, libgcc_s and libpcsclite, and adjusts the
local-top script to check for and run /usr/sbin/pcscd instead of
/sbin/pcscd

It also copies /etc/libccid_Info.plist, which is required for pcscd
to be able to load drivers.
---
 debian/initramfs/cryptopensc-hook             | 7 ++++++-
 debian/initramfs/cryptopensc-script-local-top | 4 ++--
 2 files changed, 8 insertions(+), 3 deletions(-)

diff --git a/debian/initramfs/cryptopensc-hook b/debian/initramfs/cryptopensc-hook
index 6aa30bb..2b085d0 100644
--- a/debian/initramfs/cryptopensc-hook
+++ b/debian/initramfs/cryptopensc-hook
@@ -52,7 +52,12 @@ done
 # pcscd utilizes pthread_cancel
 copy_exec /usr/sbin/pcscd
 cp -r /usr/lib/pcsc ${DESTDIR}/usr/lib
-cp /etc/reader.conf ${DESTDIR}/etc
+cp /etc/reader.conf ${DESTDIR}/etc || true
+cp /etc/libccid_Info.plist ${DESTDIR}/etc
+copy_exec /lib/$(dpkg-architecture -qDEB_HOST_MULTIARCH)/libgcc_s.so.1
+copy_exec /lib/$(dpkg-architecture -qDEB_HOST_MULTIARCH)/libusb-1.0.so.0
+copy_exec /lib/$(dpkg-architecture -qDEB_HOST_MULTIARCH)/libusb-1.0.so.0.1.0
+copy_exec /usr/lib/$(dpkg-architecture -qDEB_HOST_MULTIARCH)/libpcsclite.so.1
 
 # Install opensc commands and conf file
 copy_exec /usr/bin/opensc-tool
diff --git a/debian/initramfs/cryptopensc-script-local-top b/debian/initramfs/cryptopensc-script-local-top
index eb163f8..3bad5f4 100644
--- a/debian/initramfs/cryptopensc-script-local-top
+++ b/debian/initramfs/cryptopensc-script-local-top
@@ -20,7 +20,7 @@ esac
 
 # Hook for starting smartcard reading software
 
-if [ ! -e /sbin/pcscd ]; then
+if [ ! -e /usr/sbin/pcscd ]; then
     exit 0
 fi
 
@@ -33,4 +33,4 @@ fi
 # pcscd uses syslog which is not available (use --error or --critical to filter
 # out debug message clutter):
 #    /sbin/pcscd --error --foreground &
-/sbin/pcscd --foreground &
+/usr/sbin/pcscd --foreground &
-- 
2.13.1

